### ISSUE TITLE
[Research] CAP Training: confidence-aware parallel training for fast dLLM inference (LLaDA 2.0)

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#205]
## Background

LLaDA 2.0 introduces **CAP (Confidence-Aware Parallel) Training** as a post-training step that dramatically improves parallel decoding quality. Without CAP, aggressive parallel decoding (predicting many tokens per step) accumulates errors because the model lacks calibrated confidence. With CAP, LLaDA 2.0-flash achieves **535 tokens/sec** — 2.1× faster than equivalent AR models.

### How CAP works
During SFT, an auxiliary **confidence loss** `L_conf` is added:

```
L(θ) = L_SFT(θ) + λ * L_conf(θ)
```

`L_conf` selectively minimizes the **entropy** of `p_θ(x_0 | x_t, c)` at positions where the model predicts **correctly**. This sharpens the model's output distribution at positions it knows — without affecting uncertain predictions — enabling aggressive parallel decoding without error cascades.

### Reference
- LLaDA 2.0 tech report: `dev/repos/LLaDA2.X/tech_report.pdf`
- LLaDA 2.0 paper: https://arxiv.org/abs/2512.15745

### Current state
No CAP training exists in unturtle. The `DiffusionTrainer` compute_loss path only supports CE loss. CAP requires:
1. Identifying correctly-predicted positions (argmax match)
2. Computing entropy of the predicted distribution at those positions
3. Adding `λ * entropy` to the CE loss

## Acceptance criteria (Research phase)

- [ ] Read the LLaDA 2.0 paper section on CAP training
- [ ] Prototype `cap_lambda: float = 0.0` argument in `DiffusionTrainingArguments`
- [ ] Implement CAP auxiliary loss in `DiffusionTrainer.compute_loss`:
  ```python
  if self.cap_lambda > 0:
      probs = logits.softmax(-1)  # [B, L, V]
      predicted = probs.argmax(-1)  # [B, L]
      correct_mask = (predicted == labels) & diffusion_mask
      entropy = -(probs * probs.log().clamp_min(-100)).sum(-1)  # [B, L]
      cap_loss = entropy[correct_mask].mean()
      loss = loss + self.cap_lambda * cap_loss
  ```
- [ ] Validate that training with `cap_lambda > 0` improves parallel decoding accuracy on a held-out generation benchmark
- [ ] If validated, promote to a full implementation PR

## Context
This is a **research/prototype** issue. CAP training is primarily valuable when combined with block-decode or high-parallelism generation. Priority: **lower than CART** which is needed for basic Dream fine-tuning.

## References
- LLaDA 2.0: https://arxiv.org/abs/2512.15745
- `dev/repos/LLaDA2.X/tech_report.pdf`

## Labels
type: feat, diffusion, type: perf
